lib/plugins/budget/verify.js

```diff
@@ -55,6 +55,14 @@ module.exports = {
     // timing, request, coach etc
     for (let metricType of Object.keys(budgetForThisURL)) {
       for (let metric of Object.keys(budgetForThisURL[metricType])) {
+        // [TODO] refactor might be required
+        if (message.type === 'pagexray.pageSummary') {
+          for (let responseCode of Object.keys(message.data.responseCodes)) {
+            if (parseInt(responseCode) > 399) {
+              log.error('Error:', responseCode);
```

Comment:
Cool, I would do a couple of changes:I would log on info level since this isn't an error in sitespeed.io and then I would also fine-tune the message: Include the URL in the message so the user knows which URL that returns the error and add a short describing text.

Then I would also. move the logging to the [PageXray plugin](https://github.com/sitespeedio/sitespeed.io/tree/master/lib/plugins/pagexray) then all users will get the information if something is wrong (you don't need to run a budget).

Then keep this here but instead of logging, push it to the budget result!